launch/test/launch/test_execute_process.py

```diff
@@ -14,21 +14,39 @@
 
 """Tests for the ExecuteProcess Action."""
 
+import os
 import sys
 
 from launch import LaunchDescription
 from launch import LaunchService
 from launch.actions.execute_process import ExecuteProcess
 
+import pytest
 
-def test_execute_process_with_env():
+
+@pytest.mark.parametrize('test_input,expected', [
+    (None, [True, False]),
+    ({'TEST_NEW_ENV': '2'}, [False, True])
+])
+def test_execute_process_with_env(test_input, expected):
     """Test launching a process with an environment variable."""
-    ld = LaunchDescription([
-        ExecuteProcess(
-            cmd=[sys.executable, 'TEST_PROCESS_WITH_ENV'], output='screen',
-            env={'TEST_PROCESS_WITH_ENV': 'Hello World'},
-        )
-    ])
+    os.environ['TEST_CHANGE_CURRENT_ENV'] = '1'
+    additional_env = {'TEST_PROCESS_WITH_ENV': 'Hello World'}
+    executable = ExecuteProcess(
+        cmd=[sys.executable, 'TEST_PROCESS_WITH_ENV'],
```

Comment:
@ivanpauno it'd be nice for the test to execute something like `python -c "print('$TEST_PROCESS_WITH_ENV')"` in a shell and then check that the `stdout` of that action is as expected.